File: app/src/main/java/com/softaai/designpatterns/java/structural/adapter/MediaAdapter.java
package com.softaai.designpatterns.java.structural.adapter;

/**
 * Created by Amol Pawar on 02-04-2019.
 * softAai Apps
 */
public class MediaAdapter implements MediaPlayer{
    AdvancedMediaPlayer advancedMusicPlayer;

    public MediaAdapter(String audioType){
        if(audioType.equalsIgnoreCase("vlc")){
            advancedMusicPlayer = new VlcPlayer();
        }
        else if(audioType.equalsIgnoreCase("mp4")){
            advancedMusicPlayer = new Mp4Player();
        }
    }


    @Override
    public void play(String audioType, String fileName) {
        if(audioType.equalsIgnoreCase("vlc")){
            advancedMusicPlayer.playVlc(fileName);
        }
        else if(audioType.equalsIgnoreCase("mp4")){
            advancedMusicPlayer.playMp4(fileName);
        }
    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/structural/adapter/MediaAdapter.kt
package com.softaai.designpatterns.kotlin.structural.adapter


/**
 * Created by Amol Pawar on 02-04-2019.
 * softAai Apps
 */
class MediaAdapter(audioType: String) : MediaPlayer {
    lateinit var advancedMusicPlayer : AdvancedMediaPlayer
    init {
        if (audioType.equals("Vlc", ignoreCase = true)){
            advancedMusicPlayer = VlcPlayer()
        }
        else if (audioType.equals("Mp4", ignoreCase = true)){
            advancedMusicPlayer = Mp4Player()
        }
    }


    override fun play(audioType: String, fileName: String) {
        if (audioType.equals("Vlc", ignoreCase = true)){
            advancedMusicPlayer.playVlc(fileName)
        }
        else if (audioType.equals("Mp4", ignoreCase = true)){
            advancedMusicPlayer.playMp4(fileName)
        }
    }
}